mobile/src/components/courses/components/course-card/course-card.tsx

```diff
@@ -1,22 +1,31 @@
 import React, { FC, ReactElement } from 'react';
 
 import defaultCourseImage from '~/assets/images/default-course-image.png';
+import { AppScreenName } from '~/common/enums/enums';
 import { CourseGetResponseDto } from '~/common/types/courses/courses';
 import { Image, Pressable, Text, View } from '~/components/common/common';
 import { VendorLabel } from '~/components/courses/components/course-card/components/components';
 import { getImageUri } from '~/helpers/helpers';
+import { useAppDispatch, useAppNavigate } from '~/hooks/hooks';
+import { coursesActions } from '~/store/actions';
 
 import { styles } from './styles';
 
 type Props = {
   course: CourseGetResponseDto;
-  onCoursePress?: () => void;
 };
 
-const CourseCard: FC<Props> = ({ course, onCoursePress }): ReactElement => {
-  const { title, imageUrl, vendor } = course;
+const CourseCard: FC<Props> = ({ course }): ReactElement => {
+  const dispatch = useAppDispatch();
+  const navigation = useAppNavigate();
```

Comment:
CourseCard should be a simple component, it should not have dispatch and navigation, pass onPress via props